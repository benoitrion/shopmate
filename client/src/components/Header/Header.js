import React from 'react';

import TopAppBar from '@material/react-top-app-bar';
import MaterialIcon from '@material/react-material-icon';

import '@material/react-top-app-bar/index.scss';
import '@material/react-material-icon/index.scss';

import './Header.scss';

function Header() {
  return (
    <React.Fragment>
      <TopAppBar 
        className='top-app-bar'
        title='SHOPMATE'
        navigationIcon={<MaterialIcon
          icon='menu'
          onClick={() => console.log('click')}
        />}
        actionItems={[
          <MaterialIcon icon='search' />,
          <MaterialIcon icon='shopping_basket' />,
        ]}
      />
    </React.Fragment>
  );
}

export default Header;
