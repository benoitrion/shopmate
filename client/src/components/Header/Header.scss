@import url('https://fonts.googleapis.com/css?family=Montserrat');

.top-app-bar {
  font-family: 'Montserrat', sans-serif;
  font-size: '24px';
  font-weight: 'bold';
  color: 'FFFFFF';
}