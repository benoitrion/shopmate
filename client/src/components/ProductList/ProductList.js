import React from 'react';
import PropTypes from 'prop-types';

import { Cell, Grid, Row } from '@material/react-layout-grid';

import '@material/react-layout-grid/index.scss';

import Product from './Product/Product';

import './ProductList.scss';

function ProductList({ products }) {
  return (
    <React.Fragment>
      <Grid className="product-list_grid">
        <Row>
          {products.map(p => {
            return (
              <Cell columns={3} key={p.id}>
                  <Product product={p}/>
              </Cell>
            )
          })}
        </Row>
      </Grid>
    </React.Fragment>
  )
}

ProductList.propTypes = {
  products: PropTypes.array.isRequired
};

export default ProductList;
