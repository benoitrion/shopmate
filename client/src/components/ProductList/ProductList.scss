.product-list_grid {
  margin: 70px auto 0 auto;
  width: 1200px;
}