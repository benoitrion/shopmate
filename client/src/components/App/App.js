import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Header from '../Header/Header';
import ProductList from '../ProductList/ProductList';

import { fetchProducts } from '../../services/catalog/actions';

class App extends Component {

  constructor() {
    super();
    this.state = {
      products: []
    }
  }

  componentDidMount() {
    fetchProducts().then(data => this.setState({ products: data }));
  }

  render() {
    return (
      <React.Fragment>
        <Header></Header>
        <ProductList products={this.state.products} />
      </React.Fragment>
    );
  }
}

export default App;
