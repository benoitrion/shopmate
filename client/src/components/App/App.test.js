import { shallow } from 'enzyme';
import React from 'react';
import App from './App';
import Header from '../Header/Header';
import ProductList from '../ProductList/ProductList';

describe('App test', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<App />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
  
  it('should render the Calculator Component', () => {
    expect(wrapper.containsMatchingElement(<Header />)).toEqual(true);
    expect(wrapper.containsMatchingElement(<ProductList products={[]}/>)).toEqual(true);
  });
});