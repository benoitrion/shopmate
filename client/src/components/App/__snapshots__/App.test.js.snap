// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should render correctly 1`] = `
<Fragment>
  <Header />
  <ProductList
    products={Array []}
  />
</Fragment>
`;
